evalml/data_checks/ts_parameters_data_check.py

```diff
@@ -0,0 +1,64 @@
+"""Data check that checks whether the time series parameters are compatible with the data size."""
+from evalml.data_checks import DataCheck, DataCheckError, DataCheckMessageCode
+from evalml.utils.gen_utils import (
+    are_ts_parameters_valid_for_split,
+    contains_all_ts_parameters,
+)
+
+
+class TimeSeriesParametersDataCheck(DataCheck):
```

Comment:
Should this be a part of default data checks? Reason I'm not adding it is because it would require changing the init api of the default data checks and some of the use cases we've seen manually create their own `DataChecks` class from a list of individual `DataCheck` instances. So if they wanted to use this check, they just need to add it to their list.

Reply:
I think it could be a good idea to add it, but not necessary in this PR. Let's file an issue for it?

Reply:
Issue here: https://github.com/alteryx/evalml/issues/3125